clarifai/client/model.py

```diff
@@ -47,10 +57,243 @@ def __init__(self,
+
+    Args:
+        template (str): The template to use for the model type.
+        yaml_file (str): The yaml file to save the model params.
+
+    Returns:
+        params (Dict): Dictionary of model params for the model type.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model_params = model.get_params(template='template', yaml_file='model_params.yaml')
+    """
+    if not self.model_info.model_type_id:
+      self.load_info()
+    if self.model_info.model_type_id not in TRAINABLE_MODEL_TYPES:
+      raise UserError(f"Model type {self.model_info.model_type_id} is not trainable")
+    if template is None and self.model_info.model_type_id not in [
+        "clusterer", "embedding-classifier"
+    ]:
+      raise UserError(
+          f"Template should be provided for {self.model_info.model_type_id} model type")
+    if template is not None and self.model_info.model_type_id in [
+        "clusterer", "embedding-classifier"
+    ]:
+      raise UserError(
+          f"Template should not be provided for {self.model_info.model_type_id} model type")
+
+    request = service_pb2.ListModelTypesRequest(user_app_id=self.user_app_id,)
+    response = self._grpc_request(self.STUB.ListModelTypes, request)
+    if response.status.code != status_code_pb2.SUCCESS:
+      raise Exception(response.status)
+    params = response_to_model_params(
+        response=response, model_type_id=self.model_info.model_type_id, template=template)
+    #yaml file
+    with open(yaml_file, 'w') as f:
+      yaml.dump(params, f, default_flow_style=False, sort_keys=False)
+    #updating the global model params
+    self.training_params.update(params)
+
+    return params
+
+  def update_params(self, **kwargs) -> None:
+    """Updates the model params for the model.
+
+    Args:
+        **kwargs: model params to update.
+
+    Example:
+        >>> from clarifai.client.model import Model
+        >>> model = Model(model_id='model_id', user_id='user_id', app_id='app_id')
+        >>> model_params = model.get_params(template='template', yaml_file='model_params.yaml')
+        >>> model.update_params({param1:value1, param2:value2})
+    """
+    if self.model_info.model_type_id not in TRAINABLE_MODEL_TYPES:
+      raise UserError(f"Model type {self.model_info.model_type_id} is not trainable")
+    if len(self.training_params) == 0:
+      raise UserError(
+          f"Run 'model.get_params' to get the params for the {self.model_info.model_type_id} model type"
+      )
+
+    all_keys = [key for key in self.training_params.keys()] + [
+        key for key in self.training_params.values() if isinstance(key, dict) for key in key
+    ]
+    if not set(kwargs.keys()).issubset(all_keys):
+      raise UserError("Invalid params")
+
+    for key, value in kwargs.items():
+      find_and_replace_key(self.training_params, key, value)
```

Comment:
Could you please add some comments for this

Reply:
Sure